metadata_ui.js

```diff
@@ -0,0 +1,162 @@
+    var _create = function (opts) {
+      // Parsing the arguments passed in to find out what kind of suite is being made
+      var title, metadata, fn;
+      title = opts.args[0];
+      if (opts.args.length == 2) {
+        if (typeof opts.args[1] === 'object') {
+          metadata = opts.args[1].metadata;
+          fn = opts.args[1].tests;
+        } else if (typeof opts.args[1] === 'function') {
+          fn = opts.args[1];
+        }
+      } else if (opts.args.length == 3) { // Metadata as a param: it(title, meta, fn)
+        metadata = opts.args[1];
+        fn = opts.args[2];
+      }
+
+      // Creating the Suite object
+      var suite = Suite.create(suites[0], title);
+      suite.pending = Boolean(opts.pending);
+      suite.file = file;
+      suites.unshift(suite);
+      if (opts.isOnly) {
+        suite.parent._onlySuites = suite.parent._onlySuites.concat(suite);
+        mocha.options.hasOnly = true;
+      }
+      suite.metadata = metadata || {};
+      if (typeof fn === 'function') {
+        fn.call(suite);
+        suites.shift();
+      } else if (typeof fn === 'undefined' && !suite.pending) {
+        throw new Error('Suite "' + suite.fullTitle() + '" was defined but no callback was supplied. Supply a callback or explicitly skip the suite.');
+      }
+
+      return suite;
+    }
+
+    /**
+     * Describe a "suite" with the given `title`
+     * and callback `fn` containing nested suites
+     * and/or tests.
+     */
+
+    context.describe = context.context = function () {
+      return _create({
+        args: arguments
+      });
+    };
+
+    /**
+     * Pending describe.
+     */
+
+    context.xdescribe = context.xcontext = context.describe.skip = function (title, fn) {
+      return _create({
+        args: arguments,
+        pending: true
+      });
+    };
+
+    /**
+     * Exclusive suite.
+     */
+
+    context.describe.only = function (title, fn) {
+      return _create({
+        args: arguments,
+        isOnly: true
+      });
+    };
+
+    /**
+     * Describe a specification or test-case
+     * with the given `title` and callback `fn`
+     * acting as a thunk.
+     */
+
+    context.it = context.specify = function () {
+      var title, metadata, fn;
+
+      title = arguments[0];
+      if (arguments.length == 2) {
+        if (typeof arguments[1] === 'object') {
+          metadata = arguments[1].metadata;
+          fn = arguments[1].test;
+        } else if (typeof arguments[1] === 'function') {
+          fn = arguments[1];
+        }
+      } else if (arguments.length == 3) { // Metadata as a param: it(title, meta, fn)
+        metadata = arguments[1];
+        fn = arguments[2];
+      }
+
+      var suite = suites[0];
+      if (suite.isPending()) {
+        fn = null;
+      }
+      var test = new Test(title, fn);
+      test.file = file;
+      suite.addTest(test);
+      if (suite.metadata && metadata) {
+        var combined_metadata = dmerge(suite.metadata, metadata);
```

Comment:
is there a way we can combine this without bringing in a dependency on another module? It's generally "best practice" to avoid lots of dependencies, especially when you might not need them.